language/libra-vm/src/libra_vm.rs

```diff
@@ -639,15 +676,19 @@ impl LibraVM {
                     Value::u64(txn_max_gas_units),
                     Value::u64(txn_expiration_time),
                 ],
+                cost_strategy,
+                data_store,
+                &txn_data,
             )
             .map_err(|err| convert_prologue_runtime_error(&err, &txn_data.sender))
     }
 
     /// Run the epilogue of a transaction by calling into `EPILOGUE_NAME` function stored
     /// in the `ACCOUNT_MODULE` on chain.
-    fn run_epilogue<T: ChainState>(
+    fn run_epilogue(
         &self,
-        chain_state: &mut T,
+        data_store: &mut TransactionDataCache,
+        cost_strategy: &mut CostStrategy,
```

Comment:
Why did `run_prologue` and `run_epilogue` has argument `cost_strategy: &mut CostStrategy,` but not for `run_writeset_prologue`?

Reply:
the way the code is arranged `run_writeset_prologue` makes the cost strategy inside the function rather than taking one in. This is just what the code was (so to speak).
I think we should review the usage of the move vm inside libra at some point given the way that API is evolving, but it's for another time...